src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -80,6 +85,106 @@ enum MiscSpells
     SPELL_MAGE_CHILLED                           = 205708
 };
 
+// 110909 - Alter Time Aura
+// 342246 - Alter Time Aura
+class spell_mage_alter_time_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_alter_time_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_MAGE_ALTER_TIME_VISUAL,
+            SPELL_MAGE_MASTER_OF_TIME,
+            SPELL_MAGE_BLINK,
+        });
+    }
+
+    void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetTarget();
+        if (unit == nullptr)
+            return;
+
+        if (Player* player = unit->ToPlayer())
+        {
+            _health = player->GetHealth();
+            _loc = player->GetWorldLocation();
+        }
+    }
+
+    void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetTarget();
+        if (unit == nullptr)
+            return;
+
+        AuraRemoveMode rmmode = GetTargetApplication()->GetRemoveMode();
+        if (rmmode == AURA_REMOVE_BY_EXPIRE)
+        {
+            if (Player* player = unit->ToPlayer())
+            {
+                if (player->GetMapId() == _loc.GetMapId())
```

Comment:
I have a question - is this check even neccessary? This aura has `SpellAuraInterruptFlags::EnterWorld` which means that if player would teleport to another map, this aura will be removed with `AURA_REMOVE_BY_INTERRUPT`

Reply:
this check isn't necessary, but we may replace this one with the distance check I described below